google/cloud/pubsub_v1/publisher/_sequencer/ordered_sequencer.py

```diff
@@ -289,12 +290,12 @@ def publish(
         """
         with self._state_lock:
             if self._state == _OrderedSequencerStatus.PAUSED:
-                future = futures.Future()
+                future: Optional[futures.Future] = futures.Future()
```

Comment:
We could avoid the `Optional` and `cast` here by using a different variable name.